subsys/net/lib/nrf_cloud/src/nrf_cloud_fsm.c

```diff
@@ -530,19 +541,79 @@ static int cell_pos_cb_send(const char *const rx_buf)
 	return -EFTYPE;
 }
 
+void agps_process(const char * const buf, const size_t buf_len)
+{
+#if defined(CONFIG_NRF_CLOUD_AGPS)
+	int ret = nrf_cloud_agps_process(buf, buf_len);
+
+	LOG_DBG("A-GPS data processed; result: %d", ret);
+
+#if defined(CONFIG_NRF_CLOUD_PGPS)
+	if (ret == 0) {
+		ret = nrf_cloud_pgps_notify_prediction();
+		if (ret) {
+			LOG_ERR("Error requesting P-GPS notification: %d", ret);
+		}
+	}
+#endif
+#endif
+}
+
+void pgps_process(const char * const buf, const size_t buf_len)
+{
+#if defined(CONFIG_NRF_CLOUD_PGPS)
+	int ret = nrf_cloud_pgps_process(buf, buf_len);
+
+	LOG_DBG("P-GPS data processed; result: %d", ret);
+#endif
+}
+
+int cell_pos_process(const char * const buf)
+{
+#if defined(CONFIG_NRF_CLOUD_CELL_POS)
+	if (cell_pos_cb_send(buf) == 0) {
```

Comment:
It seems like the contents of `cell_pos_cb_send` should just be moved into this function